src/consensus/tx_verify.cpp

```diff
@@ -233,18 +235,15 @@ bool Consensus::CheckTxInputs(const CTransaction& tx, CValidationState& state, c
         }
     }
 
-    if (nValueIn < tx.GetValueOut()) {
+    const CAmount value_out = tx.GetValueOut();
+    if (nValueIn < value_out) {
         return state.DoS(100, false, REJECT_INVALID, "bad-txns-in-belowout", false,
-            strprintf("value in (%s) < value out (%s)", FormatMoney(nValueIn), FormatMoney(tx.GetValueOut())));
+            strprintf("value in (%s) < value out (%s)", FormatMoney(nValueIn), FormatMoney(value_out)));
     }
 
     // Tally transaction fees
-    CAmount nTxFee = nValueIn - tx.GetValueOut();
-    if (nTxFee < 0) {
-        return state.DoS(100, false, REJECT_INVALID, "bad-txns-fee-negative");
-    }
-    nFees += nTxFee;
-    if (!MoneyRange(nFees)) {
+    txfee = nValueIn - value_out;
+    if (!MoneyRange(txfee)) {
```

Comment:
```cpp
if (!MoneyRange(nValueIn - value_out)) {
```

Reply:
Mhmm, right, that's what we're saying in the doxygen docs now...ok, I guess.